craft_application/launchpad/models/base.py

```diff
@@ -147,7 +147,7 @@ def __setattr__(self, key: str, value: Any) -> None:  # noqa: ANN401
             return
         annotations = util.get_annotations(self.__class__)
         if key in annotations:
-            attr_path = self._attr_map.get(key, default=key)
+            attr_path = self._attr_map.get(key, key)
```

Comment:
[This line of the unit tests fail](https://github.com/canonical/craft-application/pull/615/commits/08856867f0bf7c80e51e805af915f4381dffa419#diff-cd62d08e4c10ce9b89522ebd1e498d0dd0141b63e1b87990e2930ee01da38b6cR182) with `default=`.  No idea if it's because it's typed as a mapping but is a dict in the tests or some pytest/mocking nuance.  

Regardless, I don't think there is any harm here.

Reply:
`default` only became a kwarg in [3.12](https://docs.python.org/3.12/library/stdtypes.html#dict.get). Before that it's an optional second arg. Good catch!